apps/api/src/helpers/db_operations.ts

```diff
@@ -863,3 +872,94 @@ export const copyDataToNewPackages = async (
         }
     }
 };
+
+export const findFileTreesMatchingPattern = async (
+    packageId: number,
+    pattern: string,
+): Promise<FileTree[]> => {
+    /*
+     * With the current method of creating a regex from glob patterns, the query won't find any
+     * rows if the pattern contains one of the characters in the unsupportedCharacters array.
+     * It may be possible to transform these globs to a regex that can be used in the query,
+     * but as these kinds of globs haven't been used so far, it's not a priority right now. For
+     * now, the method is set to "loop" if the pattern contains any of the unsupported characters.
+     * This means all the filetrees in the package will be fetched and checked with minimatch.
+     */
+    const unsupportedCharacters = ["[", "]", "?", "!", "|", "(", ")", "+", "@"];
+    let method = "regex";
+    if (unsupportedCharacters.some((c) => pattern.includes(c))) {
+        method = "loop";
+    }
+
+    log.debug("Using method: " + method);
+
+    const filetrees: FileTree[] = [];
+
+    /*
```

Comment:
Question: because now a pattern like `{path1/**, path2/{file1,file2}}` will be expanded to its individual patterns, could this also be used to get rid of the earlier restriction to not have PE patterns like the above? I know ORT doesn't accept them, and silently fails the PE if it has outer {}s, but what about then writing the glob as separate "sub-globs" to the PE for ORT?

Reply:
Yeah, the braces package could be used to do a similar thing for those kinds of path exclusions in the package-configuration endpoint before sending the response

Reply:
Maybe create a PR in Linear as a reminder for future improvement?

Reply:
Looks like there's already an [RFC about this](https://linear.app/double-open/issue/DO-660/rfc-handle-outer-curly-braces-in-path-exclusions) created by you